core/app/models/spree/order.rb

```diff
@@ -27,7 +27,7 @@ class CannotRebuildShipments < StandardError; end
       go_to_state :confirm
     end
 
-    self.whitelisted_ransackable_associations = %w[shipments user promotions bill_address ship_address line_items]
+    self.whitelisted_ransackable_associations = %w[shipments user order_promotions promotions bill_address ship_address line_items]
```

Comment:
The `promotions` association may still be useful in some contexts - I don't believe it's used in our application or in Solidus, but I'm hesitant to remove it unless it's also removed upstream first.